tests/xdsl_opt/test_xdsl_opt.py

```diff
@@ -1,15 +1,100 @@
 from xdsl.xdsl_opt_main import xDSLOptMain
 from contextlib import redirect_stdout
 from io import StringIO
+import pytest
+import os
 
 
 def test_opt():
-    filename = 'tests/xdsl_opt/empty_program.xdsl'
-    opt = xDSLOptMain(args=[filename])
+    opt = xDSLOptMain(args=[])
     assert list(opt.available_frontends.keys()) == ['xdsl', 'mlir']
     assert list(opt.available_targets.keys()) == ['xdsl', 'irdl', 'mlir']
     assert list(opt.available_passes.keys()) == []
 
+
+def test_empty_program():
```

Comment:
Maybe the following three tests can be parameterised

Reply:
I am not quite sure what you mean. You mean using a fancy `@pytest.mark.parametrize`? I am unfortunately not very knowledge about the possibilities this provides. Could you educate me on how it could help in this case?

Reply:
I think I understood now. let me try!

Reply:
I implemented something. Would be great to get more feedback!